### DISCUSSION TITLE
max_fanout_check_slack and max_fanout_check_limit >1e30

### DISCUSSION THREAD

**Discussion by marcopoles** [The-OpenROAD-Project/OpenROAD-flow-scripts#1563] [Q&A]
Hi, I'm importing a new 40nm tech and I have problem after cts phase. 
It finisch correctly with 23% utilization

```
==========================================================================
cts final report_design_area
--------------------------------------------------------------------------
Design area 38 u^2 23% utilization.
```

If I open the filr 4_cts.odb it looks fine:
<img width="509" alt="image" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/24896919/6d9b0d66-171e-4491-adeb-f33370e9c38a">

When the routing starts, I see following data:

```
==========================================================================
global route max_fanout_check_slack
--------------------------------------------------------------------------
1.0000000150474662**e+30**
==========================================================================
global route max_fanout_check_limit
--------------------------------------------------------------------------
1.0000000150474662**e+30**

==========================================================================
global route report_design_area
--------------------------------------------------------------------------
Design area 162 u^2 100% utilization.
```

In my lib file I have

```
    capacitive_load_unit (1,pf) ;
    voltage_unit : "1V" ;
    current_unit : "1mA" ;
    time_unit : "1ns" ;
    pulling_resistance_unit : "1kohm"; 
```

The file setTC.tcl is copied from tech nangate45 and I modified the cap values because nangate45 has capacitive_load_unit in ff:

```
set_layer_rc -layer M1 -resistance 5.4286e-03 -capacitance 7.41819E-0**5**
set_layer_rc -layer M2 -resistance 3.5714e-03 -capacitance 6.74606E-0**5**
set_layer_rc -layer M3 -resistance 3.5714e-03 -capacitance 8.88758E-0**5**
set_layer_rc -layer M4 -resistance 1.5000e-03 -capacitance 1.07121E-0**4**
set_layer_rc -layer M5 -resistance 1.5000e-03 -capacitance 1.08964E-0**4**
set_layer_rc -layer M6 -resistance 1.5000e-03 -capacitance 1.02044E-0**4**
```

My question is: which other files or lib/lef info are involved in the max_fanout calculation? 

Thanks

**Comment by marcopoles** [comment]
In the code sections, I put ** ** chars to mark in "bold" changes that I did

**Comment by maliberty** [comment]
max fanout comes from the .lib (or .sdc if manually specified).  Perhaps you have no max fanout?  Does this cause an problem beyond the reporting?

**Reply by marcopoles** [reply]
If I manually add in the .lib:

`default_max_fanout : 50;`

the max_fanout_check_slack and max_fanout_check_limit will be respectively 50 and 46, but the design area is again 100%
Does this number affect the RC calculation? If I see the routing output it looks fine:
<img width="509" alt="image" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/24896919/87f57ac2-7fc6-4369-95f2-d209eba93208">

**Reply by maliberty** [reply]
max_fanout affects optimization as it is a constraint.  That will indirectly affect things downstream.

That doesn't look like 100% utilization.  It does look like a reasonable result fwiw
